File: csharp/hamming/Hamming.cs
using System;
using System.Linq;

public static class Hamming
{
    public static int Distance(string firstStrand, string secondStrand)
    {
        firstStrand = firstStrand ?? "";
        secondStrand = secondStrand ?? "";

        if (firstStrand.Length != secondStrand.Length)
            throw new ArgumentException();

        /*
        int distance = 0;
        for (var i = 0; i < firstStrand.Length; i++)
            if (firstStrand[i] != secondStrand[i])
                distance++;

        return distance;
        */

        return firstStrand
                .Where((ch, idx) => ch != secondStrand[idx])
                .Count();
    }
}

File: clojure/hamming/src/hamming.clj
(ns hamming
;  (:require [clojure.string :as str])
)

(defn distance [strand1 strand2]
;  (when (= (count strand1) (count strand2))
;    (->> (str/split strand1 #"")
;         (map list (str/split strand2 #""))
;         (filter #(not= (first %) (second %)))
;         (count))))

;; strings are collections: > (map = "hello" "world") ;; => (false false false true false)
  (when (= (count strand1) (count strand2))
    (->> (map = strand1 strand2)
         (filter false?)
         (count))))
